OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteContext.java

```diff
@@ -149,35 +149,36 @@ public List<NetworkRouteSegment> loadRouteSegment(int x31, int y31) throws IOExc
 	}
 
 	private NetworkRoutesTile getMapRouteTile(int x31, int y31) throws IOException {
-		long tileId = getTileId(x31, y31);
+		int zm = (31 - ZOOM_TO_LOAD_TILES);
+		return getMapRouteTile15(x31 >> zm, y31 >> zm);
+	}
+
+	private NetworkRoutesTile getMapRouteTile15(int x15, int y15) throws IOException {
+		long tileId = getTileId15(x15, y15);
 		NetworkRoutesTile tile = indexedTiles.get(tileId);
 		if (tile == null) {
-			tile = loadTile(x31, y31, tileId);
+			tile = loadTile(x15, y15, tileId);
 			indexedTiles.put(tileId, tile);
 		}
 		return tile;
 	}
-	
+
 	public boolean isRouting() {
 		return routing;
 	}
-	
 
-	private NetworkRoutesTile loadTile(int x31, int y31, long tileId) throws IOException {
+	private NetworkRoutesTile loadTile(int x15, int y15, long tileId) throws IOException {
 		stats.loadedTiles++;
-		int zm = (31 - ZOOM_TO_LOAD_TILES);
-		int tileX = x31 >> zm;
-		int tileY = y31 >> zm;
-		int tileLeft = tileX << zm;
-		int tileTop = tileY << zm;
-		int tileRight = (tileX + 1) << zm;
-		int tileBottom = (tileY + 1) << zm;
 		if (routing) {
-			SearchRequest<RouteDataObject> req = BinaryMapIndexReader.buildSearchRouteRequest(tileLeft, tileRight,
-					tileTop, tileBottom, null);
+			SearchRequest<RouteDataObject> req = BinaryMapIndexReader.buildSearchRouteRequest(x15, y15);
```

Comment:
is this method needed ? refactoring goal is unclear... It looks that code before was better cause zoom was centralized.